src/login/logind-dbus.c

```diff
@@ -2210,35 +2266,63 @@ static int method_schedule_shutdown(sd_bus_message *message, void *userdata, sd_
 
         m->scheduled_shutdown_type = a;
         m->shutdown_dry_run = dry_run;
+        m->scheduled_shutdown_timeout = elapse;
 
-        if (m->nologin_timeout_source) {
-                r = sd_event_source_set_time(m->nologin_timeout_source, nologin_timeout_usec(elapse));
+        r = manager_setup_shutdown_timers(m);
+        if (r < 0)
+                return r;
+
+        r = setup_wall_message_timer(m, message);
+        if (r < 0) {
+                m->scheduled_shutdown_timeout_source = sd_event_source_unref(m->scheduled_shutdown_timeout_source);
+                return r;
+        }
+
+        r = update_schedule_file(m);
+        if (r < 0)
+                return r;
+
+        return sd_bus_reply_method_return(message, NULL);
+}
+
+static int manager_setup_shutdown_timers(Manager* m) {
+        int r;
+
+        if (m->scheduled_shutdown_timeout_source) {
+                r = sd_event_source_set_time(m->scheduled_shutdown_timeout_source,
```

Comment:
You can use `event_reset_time()`.

Reply:
Nice, that make code much shorter. What accuracy and priority do we need in that case? looks like existing users use 10 * USEC_PER_MSEC, USEC_PER_SEC or 0 as accuracy. Priority seems to be 0 always?

At this point I wonder why there is no inline documentation in headers at all :-)

Reply:
ah, 0 is DEFAULT_ACCURACY_USEC and was used before.